electrumsv/network.py

```diff
@@ -142,6 +151,11 @@ async def _synchronise_headers_for_server_tip(self, server_state: HeaderServerSt
         """
         assert app_state.headers is not None
         tip_headers = await get_chain_tips_async(server_state, self.aiohttp_session)
+        # This get_chain_tips_async call will be the first http request sent to the server.
```

Comment:
I don't understand what this code does.

Reply:
I just noticed as an aside while working on this that the `new_server_connection_event` was being set prematurely and that it instead should wait until a successful http request-response to the server so I moved the:

```
self.new_server_connection_event.set()
self.new_server_connection_event.clear()
```
To here - immediately following the initial request for server chain tip status

Reply:
I thought my code comment did explain exactly that but I will reword it to try and be clearer. This was my intent though.